src/averageAges.js

```diff
@@ -55,7 +69,19 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const children = onlyWithSon
+    ? people.filter(p => p.sex === 'm'
+      && people.some(m => m.name === p.mother))
+    : people.filter(p => people.some(m => m.name
+      === p.mother));
+
+  let sum = 0;
+
+  children.forEach((ch) => sum += ch.born - people.find(
+    (m) => m.name === ch.mother
+  ).born);
```

Comment:
this solution work, but it is difficult to understand. Addition and subtraction and search are used in one line. In addition, finding the value by the born key, which can be lost

in the future, if you see that the solution looks complicated, the best option will be to divide it into logical parts


we have to write nice and simple code.
Let me give you an example, what do you think about thsi resolve?
```javascript
 let sum = 0;

  children.forEach((ch) => {
    const mother = people.find((m) => m.name === ch.mother);
    const ageDiff = ch.born - mother.born;

    sum += ageDiff;
    }
  );
``` 

try to write the same solution with `reduce`